test/cpp/qps/qps_driver.cc

```diff
@@ -68,49 +82,124 @@ using grpc::testing::ServerType;
 using grpc::testing::RpcType;
 using grpc::testing::ResourceUsage;
 
+using std::vector;
+using std::string;
+
+vector<string> ParseCommaSeparatedIntList(string list) {
+  // Pass by value because strtok consumes its input.
+  vector<string> result;
+  char *p = strtok((char*)list.c_str(), ",");
+  while (p) {
+    result.push_back(p);
+    p = strtok(NULL, ",");
+  }
+  return result;
+}
+
+vector<ClientConfig> GetClientConfigsFromFlags() {
+  vector<ClientConfig> configs;
+  const auto rpc_types = ParseCommaSeparatedIntList(FLAGS_rpc_types);
+  const auto outstanding_rpcs_per_channel =
+      ParseCommaSeparatedIntList(FLAGS_outstanding_rpcs_per_channel);
+  const auto client_channels =
+      ParseCommaSeparatedIntList(FLAGS_client_channels);
+  const auto payload_sizes = ParseCommaSeparatedIntList(FLAGS_payload_sizes);
+  const auto client_types = ParseCommaSeparatedIntList(FLAGS_client_types);
+  const auto async_client_threads =
+      ParseCommaSeparatedIntList(FLAGS_async_client_threads);
+
+  ClientConfig cc;
+  cc.set_enable_ssl(FLAGS_enable_ssl);
+
+  for (const auto& rpc_type_str : rpc_types) {
```

Comment:
I'd like to see some work go in to making it easier to add a new argument here.

Currently:
- update flags
- update protobuf
- understand this loop and extend it

Part 3 is a sticking point for me: it'll become very fragile over time.

Could we (for example):
Write a mixer function that takes map<flag:string, values:string> --> vector<map<flag:string, value:string>>, and do the loop work in that, so that we can (simply) populate the map and then parse the result into protobufs here?

Is this even the right approach though?

Would it be better to create driver .cc files for each scenario set that we're interested in, and leave this program as a simple one off runner?